batch/batch/worker.py

```diff
@@ -606,67 +607,69 @@ def id(self):
         return (self.batch_id, self.job_id)
 
     async def run(self, worker):
-        run_start_time = time_msecs()
+        async with worker.cpu_sem(self.cpu_in_mcpu, f'{self}'):
+            self.start_time = time_msecs()
```

Comment:
The code is fine, you don't need to do anything here, but I wonder if could generalize the ContainerStepManager to also track and time the steps of the Job (including the total runtime).

Reply:
I don't see what you mean exactly. We're storing the timing of each step. The total duration is computed from the start and end time.

Reply:
I mean using a Python ContextManager to manage and time the changes of state, which are then included in the status.  So writing something like (which we do in Container):

```
            async with self.step('creating'):
```

instead of this, which we're doing in Job:

```
            self.start_time = time_msecs()
            ...
            log.info(f'{self}: initializing')
            self.state = 'initializing'
```

So then we could have a runtime block for the Job like we did before, and we'll get timing for job steps (as well as container steps).

Again, not really a big deal.